File: coyote/models/msg_namespace_get_out.py
# this file is @generated
from pydantic import Field
from datetime import datetime

from ..internal.base_model import BaseModel

from .retention import Retention
from .storage_type import StorageType


class MsgNamespaceGetOut(BaseModel):
    name: str

    retention: Retention

    storage_type: StorageType = Field(alias="storage_type")

    created: datetime

    updated: datetime

File: src/models/msg_namespace_get_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::retention::Retention;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct MsgNamespaceGetOut {
    pub name: String,

    pub retention: Retention,

    pub created: jiff::Timestamp,

    pub updated: jiff::Timestamp,
}

impl MsgNamespaceGetOut {
    pub fn new(
        name: String,
        retention: Retention,
        created: jiff::Timestamp,
        updated: jiff::Timestamp,
    ) -> Self {
        Self {
            name,
            retention,
            created,
            updated,
        }
    }
}
